### PR TITLE
quic: add experimental-quic build flag

### PR DESCRIPTION
**Pull request by jasnell**
Fixes: https://github.com/nodejs/quic/issues/55

##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: src/node_metadata.cc [hunk 1]
```diff
@@ -10,7 +10,9 @@
 
 #if HAVE_OPENSSL
 #include <openssl/opensslv.h>
-#include <ngtcp2/ngtcp2.h>
+#if defined(NODE_EXPERIMENTAL_QUIC)
+#include <ngtcp2/version.h>
```
Comment (trivikr) [inline src/node_metadata.cc]: ?
```suggestion
#include <ngtcp2/ngtcp2.h>
```

Comment (jasnell) [reply src/node_metadata.cc]: either works, version makes it more explicit what we're pulling in here

**Comment by jasnell** [comment]
Hmm... not working on linux...

**Comment by jasnell** [comment]
heh... doh, forgot about the cctests :-)

**Comment by jasnell** [comment]
@nodejs/quic ... be aware, I'm going to land this now. Once it lands, you'll need to use `./configure --experimental-quic` on Linux and `vcbuild experimental-quic` to build with the QUIC support enabled.

**Comment by jasnell** [comment]
landed

**Pull request closed without merging**